Admin redownsample

derek-pryor opened the pull request:
Modification that allows an admin (Keycloak `admin` role / Django `is_staff` flag) to request a re-downsample of a channel with a different coordinate frame. This appears as a new button next to the current downsample buttons on the channel page in the management console.

In addition after any channel is downsampled the redis cache is cleared of any keys from that channel so that we don't accidentally continue serve up stale data after it has been changed.

Testing has not been completed but will be before the PR is merged.

movestill commented on django/bossspatialdb/views.py:
```diff
@@ -292,6 +295,19 @@ def get(self, request, collection, experiment, channel):
+                try:
+                    cache = RedisKVIO(settings.KVIO_SETTINGS)
+                    pipe = cache.cache_client.pipeline()
+                    for key in pipe.scan_iter("CACHED-CUBOID&"+lookup_key+"&*"):
```
In `kvio.py`, `generate_cached_cuboid_keys()`, it looks like you could also have this prefix in Redis: `CACHED-CUBOID&ISO&`. We should clear those, too.

movestill replied:
Actually, thinking about a little more, how about creating a method to clear these keys in RedisKVIO itself? Then all the code related to the working with cache keys is closer together (file-wise).

derek-pryor replied:
Since this is more like a temporary hack, that will be removed once the longer term solution is implemented, I wanted to keep the changes contained.
In the longer term solution there will be other changes to spdb so this can be rolled into that.

movestill replied:
Ok fair enough.

derek-pryor commented:
Testing complete. Fixed a bug in the cache clearing code and added a note about moving the code into spdb when the next downsample update is implemented.
